src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -131,6 +136,36 @@ public static Variables GetVariableValues(Document document, ISchema schema, Var
         private static void ThrowNullError(string variableName)
             => throw new InvalidVariableError(variableName, "Received a null input for a non-null variable.");
 
+        private struct VariableName
+        {
+            public VariableName(VariableName variableName, int index)
+            {
+                Name = variableName;
+                Index = index;
+                ChildName = null;
+            }
+            public VariableName(VariableName variableName, string childName)
+            {
+                if (variableName.ChildName == null)
+                {
+                    Name = variableName.Name;
+                    Index = variableName.Index;
+                }
+                else
+                {
+                    Name = variableName;
```

Comment:
So here will be concatenation via `ToString` on second (and deeper) level of depth?

Reply:
yup